src/renderer/port-forward/port-forward-store/port-forward-store.ts

```diff
@@ -9,19 +9,22 @@ import type { StorageLayer } from "../../utils";
 import { autoBind, disposer } from "../../utils";
 import type { ForwardedPort } from "../port-forward-item";
 import { PortForwardItem } from "../port-forward-item";
-import { apiBase } from "../../api";
 import { waitUntilFree } from "tcp-port-used";
 import logger from "../../../common/logger";
+import type { JsonApi } from "../../../common/k8s-api/json-api";
+import type { RequestActivePortForward } from "./request-active-port-forward.injectable";
 
 interface Dependencies {
-  storage: StorageLayer<ForwardedPort[] | undefined>;
+  readonly storage: StorageLayer<ForwardedPort[] | undefined>;
   notifyErrorPortForwarding: (message: string) => void;
+  readonly apiBase: JsonApi; // TODO: replace with individual dependencies
+  requestActivePortForward: RequestActivePortForward;
 }
 
 export class PortForwardStore extends ItemStore<PortForwardItem> {
   @observable portForwards: PortForwardItem[] = [];
 
-  constructor(private dependencies: Dependencies) {
+  constructor(protected readonly dependencies: Dependencies) {
```

Comment:
Hmm. Why protected?

Reply:
This one is easy to answer because in general classes can be extended and those child classes can have dependencies too and it is best practise if they can all use the same `dependencies` field so for dependencies they should be `protected`. This is easier to see in places like `src/renderer/components/dock/install-chart/store.ts`

Reply:
But would it not be a better policy to prefer encapsulation over visibility for speculation? To make properties more visible means a curious coder will try to find out why something is protected, find nothing, and be perplexed.

Here I see no reason to not wait for the first inheritor to require protected dependencies, and only then adjust.

Interesting conversation, but pretty trivial here. I suspect somewhere else it might not be :) Resolved.